### ISSUE TITLE
Failed to copy ... Insufficient permissions to access this file

### ISSUE THREAD

**Issue by karimmohraz** [minio/minio#5950]
cannot upload recursively folder to S3
however, mc mirror works fine, and other go-client successfully uploaded same data.
cp 1 file works also.
works with linux client RELEASE.2018-03-25T01-22-22Z

## Expected Behavior

used to work fine with earlier 2017? versions of mc

## Current Behavior
```
<!--- If describing a bug, tell us what happens instead of the expected behavior -->
<!--- If suggesting a change/improvement, explain the difference from current behavior -->
mc: <DEBUG> PUT /data/test/Adidas/0.jpg HTTP/1.1
Host: 7579be90.files.xxx
User-Agent: Minio (darwin; amd64) minio-go/6.0.1 mc/2018-04-28T00:08:20Z
Content-Length: 28078
Authorization: AWS4-HMAC-SHA256 Credential=xxx/xxx/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date;x-amz-meta-com.apple.quarantine, Signature=**REDACTED**
Content-Type: image/jpeg
X-Amz-Content-Sha256: UNSIGNED-PAYLOAD
X-Amz-Date: 20180517T170510Z
X-Amz-Meta-Com.apple.quarantine: 0081;5a7d82f0;Google\x20Chrome.app;4C01FC7F-B3E1-41B1-A42A-CBE5FA6548F4
Accept-Encoding: gzip

mc: <DEBUG> HTTP/1.1 400 Bad Request
Content-Length: 291
Accept-Ranges: bytes
Connection: keep-alive
Content-Type: application/xml
Date: Thu, 17 May 2018 17:05:10 GMT
Server: nginx/1.13.3
Vary: Origin
X-Amz-Request-Id: 152F7D6E1E5FADB3

<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>AccessDenied</Code><Message>There were headers present in the request which were not signed</Message><Key></Key><BucketName></BucketName><Resource>/data/test/Adidas/0.jpg</Resource><RequestId>3L137</RequestId><HostId>3L137</HostId></Error>mc: <DEBUG> Response Time:  16.234717ms

mc: <ERROR> Failed to copy `test/Adidas/0.jpg`. Insufficient permissions to access this file `https://7579be90.files.xxx/data/test/Adidas/0.jpg`
 (3) cp-main.go:395 cmd.doCopySession(..) Tags: [test/Adidas/0.jpg]
 (2) common-methods.go:180 cmd.uploadSourceToTargetURL(..) Tags: [https://7579be90.files.xxx/data/test/Adidas/0.jpg]
 (1) common-methods.go:114 cmd.putTargetStream(..) Tags: [mlV2train60, https://7579be90.files.xxx/data/test/Adidas/0.jpg]
 (0) client-s3.go:655 cmd.(*s3Client).Put(..)
 Release-Tag:RELEASE.2018-04-28T00-08-20Z | Commit:1b794a4ae1af | Host:ZZZM34100218A | OS:darwin | Arch:amd64 | Lang:go1.10.1 | Mem:3.3MB/13MB | Heap:3.3MB/7.3MB
```

## Steps to Reproduce (for bugs)

1. mc cp -r test/Adidas/ mlV2train60/data/test/
2.
3.
4.

## Your Environment

* Version used (`minio version`): 2018-04-28T00:08:20Z and 2018-03-25 (installed with wget)
* Environment name and version (e.g. nginx 1.9.1):
* Server type and version:
* Operating System and version (`uname -a`):
Darwin 100218A 17.5.0 Darwin Kernel Version 17.5.0: Mon Mar  5 22:24:32 PST 2018; root:xnu-4570.51.1~1/RELEASE_X86_64 x86_64
* Link to your project:

**Comment by harshavardhana** [comment]
This issue is a variant of https://github.com/minio/minio/issues/5857

**Comment by kannappanr** [comment]
Fix for #5857 will also fix this issue.
